pkg/infra/iac2/templates/secret_version/factory.ts

```diff
@@ -0,0 +1,25 @@
+import * as aws from '@pulumi/aws'
+import * as fs from 'fs'
+
+interface Args {
+    SecretName: string
+    Secret: aws.secretsmanager.Secret
+    Path: string
+}
+
+// noinspection JSUnusedLocalSymbols
+function create(args: Args): void {
+    return fs.readFile(args.Path, (err, data) => {
+        if (err != null) {
+            return
+        }
+        new aws.secretsmanager.SecretVersion(
+            args.SecretName,
+            {
+                secretId: args.Secret.id,
+                secretBinary: data.toString('base64'),
+            },
+            { protect: this.lib.protect }
+        )
+    })
```

Comment:
does anything depend on this in deployment order? Because the callback will be invoked async and we wont have the actual secretversion to reference since it wont be assigned to a constant

Reply:
no — this depends on Secret, but nothing depends on this. And if you look at the return signature, it's void anyway.

Which btw will cause a bit of a weird (but valid) ts right now, because we'll actually assign a variable to it. Maybe I should add an exception that if the return type is void, don't assign it a var.

Reply:
> Maybe I should add an exception that if the return type is void, don't assign it a var.

Filed as #450. I put it in milestone 4, but it's easy, so I may just do it one afternoon.